### ISSUE TITLE
Cannot pass "orm=true" to paging during Cursor pagination

### ISSUE THREAD

**Issue by HDM007** [uriyyo/fastapi-pagination#623]
I'm trying to implement Cursor-based pagination, using sqlalchemy.ext pagination function, CursorPage, and other guidance from the docs.

When I follow the docs to implement cursor pagination and send a request, I receive this error:

```
raise ValueError("If orm=False then q cannot be a Query")
ValueError: If orm=False then q cannot be a Query
```

I've made sure my models have Config class with orm_mode = True, but I am not able to influence the actual paging orm mode. Is there a workaround for this that doesn't require a completely custom solution?

**Comment by uriyyo** [comment]
Hi @HDM007,

Did you use an example from docs or did you slightly change it?

I updated example, could you please check again?

**Comment by HDM007** [comment]
Hi @uriyyo ,
I did an exact copy of the example that was up on the Cursor page, linked here:
https://uriyyo-fastapi-pagination.netlify.app/tutorials/cursor-pagination/ 

When I try again, after wrapping app with `add_pagination()`:
1.) I import these key pieces:

```
from fastapi_pagination import add_pagination
from fastapi_pagination.cursor import CursorPage
from fastapi_pagination.ext.sqlalchemy import paginate
```

2.) I define an async function on a route, and set the response model to be `CursorPage[MySchema]` -- MySchema is 1-dimensional, e.g. no nested lists, dicts, or similarly complex data structures, and I've enabled `class Config: orm_mode = True`:

```
@router.get("/{key}/item", dependencies=[Depends(auth.check_user)])
async def get_property_guests(key: str, user: User = Depends(auth.check_user), db: Session = Depends(get_db) -> CursorPage[MySchema]:
```

3.) In the function body, I call paginate on a query:

```
query = await crud.my_query_to_paginate
return paginate(query)
```

OR

`return paginate(db, query)`

4.) I now receive the following error:

```
raise RuntimeError("Use params or add_pagination")
RuntimeError: Use params or add_pagination
```

5.) Alternatively, when I explicitly pass `response_model=CursorPage[MySchema]` to the route, I get the following error:

```
raise ValueError("If orm=False then q cannot be a Query")
ValueError: If orm=False then q cannot be a Query
```

When I switch to using limit-offset, I am able to get the paginated results without issue. Is this information useful to you? What else do you recommend?

In the short-term, I intend to stick with limit-offset, but near future, I need to either figure out how to get fastapi-pagination cursor working, or I need to build my own cursor pagination.

**Comment by uriyyo** [comment]
@HDM007 Could you please provide a minimal code snippet so I can reproduce this issue and debug it?
Currently, I can't reproduce it(

**Comment by HDM007** [comment]
Can do -- will put together an example and send it up to you in the next couple of days, need to pick a good endpoint to use for an example.

**Comment by HDM007** [comment]
Hi again @uriyyo :

I am no longer able to replicate the `orm=true` error, believe it or not. I think perhaps I was passing object instances to the Pagination function? I know it arose when I wasn't using `params` to indicate a cursor and a size, so perhaps I split off at that error and wound up calling objects. Perhaps passing params is explicitly required in Cursor pagination?

This below screenshot is where I landed. Cursor seems to be working (e.g. if I say `params.size = 2` in the code and I send a request where `?page=2` then I am able to get the correct set of results). 

The issue I now face is pictured in the Postman JSON screenshot -- there are no next_page or previous_page fields in response. Do these need to be explicitly declared, should they be cursor items? Is it like that because the user should just pass up `page=int` and the package will handle identifying the pointer without querying the whole table? Do I need to ignore it entirely? I think this is mainly my own misunderstanding on what to expect with the results of the package.

Screenshot 2023-04-27 at 12.55.24 PM.png
Screenshot 2023-04-27 at 1.41.16 PM.png

**Comment by uriyyo** [comment]
Hi @HDM007,

It looks like screenshots were not uploaded(

For cursor pagination, you need to have `order_by` clause specified.

**Comment by HDM007** [comment]
Code Sample and Response Below!

Note: 
within app.main.py, I've added pagination to the app using add_pagination()

**MyRouter.py**
```
from fastapi_pagination import Params
from fastapi_pagination.cursor import CursorPage
from fastapi_pagination.ext.sqlalchemy import paginate

#standard scaffolding
get_db = Util.get_db
router = APIRouter()

@router.get("/", response_model=CursorPage[ItemBase])
async def get_items(db: Session = Depends(get_db), params: Params = Depends()):
  items = paginate(conn=db, query=db.query(Item), params=params)
  return items
```

**Response**
```
{
  "items": [
    {itemModel},
    {itemModel}
  ],
  "previous_page": null,
  "next_page": null
}
```

I'll go ahead and add `order_by`

**Comment by uriyyo** [comment]
Hi @HDM007,

Any updates? Can I somehow help you?

**Comment by HDM007** [comment]
Hey Yuri! It looks like I was either passing the query incorrectly (perhaps had a malformed query object, which in turn wasn't accepted by paginate) or I was not using transformer correctly (which I am now using to pass a ModelUtil to build objects form Models).

I will circle back with a summary for you and send up some notes on how I got it working + what worked for me. Thanks for checking in, I completely forgot to follow up!

**Comment by HDM007** [comment]
Yuri--
The key that was throwing my responses off was the way I implemented response_model.

In my original implementation, I was explicitly declaring response_model=CursorPage[MyModel].

I went from this:
@route.post("/route/path", response_model=CursorPage[MyModel])
def: do_something(variable1: str, variable2: str):

which would net results like I mentioned, with next, previous, and limit.

To this:
@route.post("/route/path")
def do_something(variable1: str, variable2: str, params: Params = Depends()) -> CursorPage[MyModel]

which would net results with total, page, size, and pages. With the second approach, passing the page number in the query_params with ?page=2, for instance, yielded fantastic results.

Adding a transformer with a function to map my SQLAlchemy models to the response really made this sing. 

I suspect my original issue stemmed from passing the SQLAlchemy queries incorrectly. I've made a point to pass queries instead of actual query results anywhere where I use cursor pagination.

With this fix, I certainly see myself using the package to a greater extent in near future.
